### ISSUE TITLE
--print-build-logs with @nix actions

### ISSUE THREAD

**Issue by jkarni** [NixOS/nix#6826]
**Is your feature request related to a problem? Please describe.**
[garnix](https://garnix.io) parses the `@nix` lines (e.g. `@nix { "action": "setPhase", "phase": "unpackPhase" }`) output by `nix log` to figure out what phases logs belong to. We now want to stream the logs, but found out that the output of `nix build --print-build-logs` (or any of the `log-format`s) is actually *different* than the output of `nix log`, and doesn't contain these phases (*even* `internal-json`!). 

**Describe the solution you'd like**
A `log-format` that outputs phase actions. This would allow tooling to be phase-aware (in much the same way the progress bar is)

**Describe alternatives you've considered**
I've even considered hacking through the ANSI escape codes to get to the phase.

**Comment by thufschmitt** [comment]
It is actually present in the `internal-json` format, it's just that that format is a bit hard to parse :)

The log lines corresponding to `setPhase` are the `result` lines of type `104` (coming from [here](https://github.com/NixOS/nix/blob/73fde9eed06dfdef5d37b3d798cfc98a542a4d73/src/libutil/logging.hh#L32)):

```
@nix {"action":"result","fields":["unpackPhase"],"id":418969764757508,"type":104}
```

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
